internal/api/graphql/access/token_auth.go

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package access
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type TokenAuth struct {
+	logger Logger
+	secret []byte
+}
+
+func NewTokenAuth(l Logger) *TokenAuth {
+	return &TokenAuth{logger: l, secret: []byte(os.Getenv("AUTH_TOKEN_SECRET"))}
+}
+
+type TokenClaims struct {
+	Name    string `json:"name"`
+	Role    string `json:"role"`
+	Version string `json:"version"`
+	jwt.RegisteredClaims
+}
+
+func (ta *TokenAuth) GetMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		tokenString := c.GetHeader("Authorization")
+
+		if tokenString == "" {
+			ta.logger.Error("Trying to use API without authorization header")
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Authorization header is required"})
+			c.Abort()
+			return
+		}
+
+		token, claims, err := ta.parseFromString(tokenString)
+		if err != nil {
+			ta.logger.Error("JWT parsing error: ", err.Error())
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Token parsing error"})
+			c.Abort()
+			return
+		} else if !token.Valid {
```

Comment:
I'm not sure about this one but maybe we should also check for expiration (date) explicitly. Something like: 

```.go
		if claims.ExpiresAt.Before(time.Now()) {
			ta.logger.Warn("Expired token")
			c.JSON(http.StatusUnauthorized, gin.H{"error": "Token expired"})
			c.Abort()
			return
		}
```

Reply:
Done